src/Arcus.BackgroundJobs.DataBricks/DatabricksJobMetrics.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Arcus.Messaging.Pumps.Abstractions;
+using Arcus.Security.Core;
+using GuardNet;
+using Microsoft.Azure.Databricks.Client;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.BackgroundJobs.Databricks
+{
+    /// <summary>
+    /// Representing a background job that repeatedly queries the Databricks instance for finished job runs and writes the report as a metric.
+    /// </summary>
+    public class DatabricksJobMetrics : MessagePump
```

Comment:
We are not using a message pump here but simple a process to run every x min/hours; see registration where we need to move to a scheduled job approach

Reply:
Yes, done that. Required a writable field to keep track of the previous time window though.